merlin/models/tf/models/base.py

```diff
@@ -374,6 +374,10 @@ def _create_metrics(self, metrics=None):
         else:
             out = metrics
 
+        for metric in tf.nest.flatten(out):
+            # We need to ensure metrics passed to `compile()` are reset
+            if metric and metric.built:
+                metric.reset_states()
```

Comment:
Based on [this code](https://github.com/keras-team/keras/blob/v2.10.0/keras/metrics/base_metric.py#L419-L424), looks like `reset_states` has been deprecated. Replaced by `reset_state`. This works, since it's an alias for backwards compatibility, but maybe for consistency with the latest API, can be renamed here.